### PR TITLE
CES-46 Delete an occurrence from Powwow tables

### PR DESCRIPTION
**Pull request by VuHoDac**
https://issues.liferay.com/browse/CES-46

### CODE REVIEW

File: modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/meetings/portlet/MeetingsPortlet.java [hunk 1]
```diff
@@ -151,6 +152,55 @@ public void deletePowwowMeeting(
...
+						occurrenceId, OccurrenceStatus.DELETE);
+
+				if (occurrence.getCalendarBookingId() > 0) {
+					CalendarBooking calendarBooking =
+						CalendarBookingLocalServiceUtil
+							.fetchCalendarBooking(occurrence.getCalendarBookingId());
+
+					calendarBooking.setStatus(CalendarBookingWorkflowConstants.STATUS_IN_TRASH);
+
+					CalendarBookingLocalServiceUtil.updateCalendarBooking(calendarBooking);
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/meetings/portlet/MeetingsPortlet.java]: Again, do not update the object directly this way. We MUST call the provided method from service layer `CalendarBookingLocalServiceUtil.updateStatus`

File: modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/meetings/portlet/MeetingsPortlet.java [hunk 2]
```diff
@@ -1097,6 +1147,39 @@ protected void updateOccurrence(
...
+			themeDisplay.getCompanyId(), powwowParticipants, serviceContext);
+
+		Map<Locale, String> titleMap =
+			getLocalizationMap(powwowMeeting.getName());
+
+		Map<Locale, String> descriptionMap = getLocalizationMap(
+			getCalendarBookingDescription(calendarBooking.getDescription(),
+				powwowMeeting, serviceContext));
+
+		_updateCalendarBookingExceptionDate(powwowMeetingOccurrence,
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/meetings/portlet/MeetingsPortlet.java:1130]: rebase CES-34, then call with updated method

File: modules/apps/powwow/powwow-portlet/docroot/meetings/occurence_row_action.jsp [hunk 3]
```diff
@@ -14,31 +14,95 @@
...
 --%>
 
+<%@page import="com.liferay.portal.kernel.portlet.LiferayWindowState"%>
 <%@ include file="/init.jsp" %>
 
 <%
 ResultRow row = (ResultRow)request.getAttribute(WebKeys.SEARCH_CONTAINER_RESULT_ROW);
 
 PowwowMeetingOccurrence powwowMeetingOccurrence = (PowwowMeetingOccurrence) row.getObject();
+
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/meetings/occurence_row_action.jsp:23]: please combine with my adjustment: completed and deleted

File: modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/meetings/portlet/MeetingsPortlet.java [hunk 4]
```diff
@@ -151,6 +152,55 @@ public void deletePowwowMeeting(
...
+					.updateOccurrenceStatus(
+						occurrence.getPowwowMeetingId(),
+						occurrenceId, OccurrenceStatus.DELETE);
+
+				if (occurrence.getCalendarBookingId() > 0) {
+					CalendarBooking calendarBooking =
+						CalendarBookingLocalServiceUtil
+							.fetchCalendarBooking(occurrence.getCalendarBookingId());
+
+					calendarBooking.setStatus(CalendarBookingWorkflowConstants.STATUS_IN_TRASH);
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/meetings/portlet/MeetingsPortlet.java]: consider `STATUS_IN_TRASH` or `STATUS_INACTIVE`? just wondering, as long as you feel it's correct.

**Review by hhtank — CHANGES_REQUESTED** [review]

**Comment by VuHoDac** [comment]
updated

**Pull request closed without merging**